### PR TITLE
Feature/UI for choosing selftest as an option

### PR DESCRIPTION
**Pull request by Juliehoejrisp**
Created a new page for registration of selftest and normal test.

### CODE REVIEW

File: NDB.Covid19/NDB.Covid19.Droid/Utils/AuthErrorUtils.cs [hunk 1]
```diff
@@ -14,36 +14,41 @@ public static class AuthErrorUtils
...
-            GoToErrorPage(parent, REGISTER_ERROR_FETCH_FHI_DATA_HEADER, REGISTER_ERROR_FETCH_FHI_DATA_DESCRIPTION, REGISTER_ERROR_DISMISS);
+            GoToErrorPage(parent, REGISTER_ERROR_FETCH_FHI_DATA_HEADER, REGISTER_ERROR_FETCH_FHI_DATA_DESCRIPTION, REGISTER_ERROR_DISMISS, REGISTER_SELFTEST_BUTTON);
             LogUtils.LogException(severity, e, errorMessage);
         }
 
-        public static void GoToErrorPage(Activity parent, string title, string description, string button, string subtitle = null)
+       
+
+
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.Droid/Utils/AuthErrorUtils.cs]: Delete empty lines

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/ErrorActivities/GeneralErrorActivity.cs [hunk 2]
```diff
@@ -83,6 +84,13 @@ void Init()
             }).Run;
 
             this.Title = textFieldsBundle.GetString("title");
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/ErrorActivities/GeneralErrorActivity.cs]: Delete empty line

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/ErrorActivities/GeneralErrorActivity.cs [hunk 3]
```diff
@@ -83,6 +84,13 @@ void Init()
...
+
+            Button registerSelftestButton = FindViewById<Button>(Resource.Id.continue_button);
+            registerSelftestButton.Text = continueButtonText;
+            registerSelftestButton.ContentDescription = ErrorViewModel.REGISTER_SELFTEST_BUTTON;
+            button.ContentDescription = continueButtonText;
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/ErrorActivities/GeneralErrorActivity.cs]: Delete empty line

File: NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs [hunk 4]
```diff
@@ -107,8 +107,17 @@ private void SetupStyling()
...
-        }
 
+
+			StyleUtil.InitButtonStyling(ContinueWithSelftestButton, ErrorViewModel.REGISTER_SELFTEST_BUTTON);
+
+			if (ErrorTitle != ErrorViewModel.REGISTER_ERROR_NOMATCH_HEADER)
+            {
+				ContinueWithSelftestButton.Hidden = true;
+			}
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs]: Delete empty line

File: NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs [hunk 5]
```diff
@@ -107,8 +107,17 @@ private void SetupStyling()
             {
                 ErrorMessageLabel.AccessibilityLabel = ErrorViewModel.REGISTER_ERROR_ACCESSIBILITY_TOOMANYTRIES_DESCRIPTION;
             }
-        }
 
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs]: Delete empty line

File: NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        UIViewController _authViewController;
+        UIButton _ContinueButton;
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+            _viewmodel = new SelftestRegistrationViewModel();
+            SetTexts();
+            SetupStyling();
+            //ContinueWithMSISBtn.TouchUpInside += ContinueButtonTapped;
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs]: Why is this commented out?

File: NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+            Header.SizeToFit();
+        }
+
+        partial void CloseBtn_TouchUpInside(UIKit.UIButton sender)
+        {
+            LeaveController();
+        }
+
+        partial void ContinueWithMSISBtn_TouchUpInside(UIKit.UIButton sender)
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs]: Delete empty line

File: NDB.Covid19/NDB.Covid19/ViewModels/SelftestRegistrationViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    public static string POSTIVE_TEST_SELFTESTBUTTON_TEXT => "POSTIVE_TEST_SELFTESTBUTTON_TEXT".Translate();
+    public static string HEADER_LABEL_SELFTEST_REGISTRATION_TEXT => "HEADER_LABEL_SELFTEST_REGISTRATION_TEXT".Translate();
+
+
+
+
+      public SelftestRegistrationViewModel() { }
+
+
+    }
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/ViewModels/SelftestRegistrationViewModel.cs]: Delete empty lines and fix indentation

**Review by pavshr — CHANGES_REQUESTED** [review]
Good, but fix some small stuff first

File: NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs (hunk 6 above)
Comment (Juliehoejrisp) [reply NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs]: i forgot to delete it, but it is not supposed to be in the code. I have deleted it now.

File: NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        public static SelftestOptionViewController GetSelftestOptionViewController()
+        {
+            UIStoryboard storyboard = UIStoryboard.FromName("SelftestOption", null);
+            SelftestOptionViewController vc = storyboard.InstantiateInitialViewController() as SelftestOptionViewController;
+            vc.ModalPresentationStyle = UIModalPresentationStyle.FullScreen;
+            return vc;
+        }
+
+        SelftestRegistrationViewModel _viewmodel;
+        UIViewController _authViewController;
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs]: Why do you need this one?

File: NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        UIButton _ContinueButton;
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/SelftestOption/SelftestOptionViewController.cs]: Is this one needed as well?

File: NDB.Covid19/NDB.Covid19/ViewModels/SelftestRegistrationViewModel.cs [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace NDB.Covid19.ViewModels
+{
+    public class SelftestRegistrationViewModel
+    {
+        
+    public static string POSITIVE_TEST_MSISBUTTON_TEXT => "CHOOSE_CONFIRMED_TEST_BUTTON_TEXT".Translate();
+    public static string POSTIVE_TEST_SELFTESTBUTTON_TEXT => "CHOOSE_SELF_TEST_BUTTON_TEXT".Translate();
+    public static string HEADER_LABEL_SELFTEST_REGISTRATION_TEXT => "CHOOSE_TYPE_OF_TEST_HEADER".Translate();
+    public SelftestRegistrationViewModel() { }
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/ViewModels/SelftestRegistrationViewModel.cs]: Add an extra linebreak between strings and this constructor method

**Review by pavshr — CHANGES_REQUESTED** [review]

**Review by pavshr — APPROVED** [review]

**Pull request merged**